### PR TITLE
Feature/se 46

### LINKED ISSUE #46: Create the Front-end for Employee Creation
image

### PR DESCRIPTION
**Pull request by Arkcorner**
closes SE2-WS23/documentation#46 

git meme

### CODE REVIEW

File: src/pages/EmployeeCreation/EmployeeCreationPage.jsx
Comment (Chrissydrx) [inline src/pages/EmployeeCreation/EmployeeCreationPage.jsx:1]: Layout is broken for small devices. Avatar is stretched.
image

Comment (Chrissydrx) [inline src/pages/EmployeeCreation/EmployeeCreationPage.jsx:1]: You don't use our prettier settings

File: src/pages/EmployeeCreation/EmployeeCreationPage.jsx [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+  };
+  const handleEmployeeAddressChange = (event) => {
+    setEmployeeAddress(event.target.value);
+  };
+  const handleEmployeeExtraInfoChange = (event) => {
+    setEmployeeExtraInfo(event.target.value);
+  };
+ const handleChangeAvatar = (event) => {
+    console.log("Avatar has been changed");
+  }
```
Comment (Chrissydrx) [inline src/pages/EmployeeCreation/EmployeeCreationPage.jsx]: Please add space between those functions

File: src/pages/EmployeeCreation/EmployeeCreationPage.jsx [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+            justifyContent: "center",
+            alignItems: "center",
+            minHeight: "100vh",
+            minWidth: "100vw",
+            flexDirection: "column",
+          }}
+        >
+          <Box maxWidth="50%">
+            <Box>
+              <Avatar sx={{ width: "10vw", height: "20vh", variant: "circular" }} />
```
Comment (Chrissydrx) [inline src/pages/EmployeeCreation/EmployeeCreationPage.jsx]: Weird width and height settings for avatar

File: src/pages/EmployeeCreation/components/UploadBox.jsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
+import React, { useState } from "react";
+import { Button, Box, Input } from "@mui/material";
+import UploadedFile from "./UploadedFile";
+
+  /**
+  * @returns {JSX.Element} The Box holding all uploaded files in the Employee creation page component.
+  */
+
+function UploadBox() {
+  const [selectedFiles, setSelectedFiles] = useState([]);
```
Comment (Chrissydrx) [inline src/pages/EmployeeCreation/components/UploadBox.jsx]: That state can not be accessed in your page component

File: src/pages/EmployeeCreation/components/UploadBox.jsx [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+  const handleFileSubmit = (event) => {
+    const files = event.target.files;
+    const newFiles = Array.from(files);
+
+    setSelectedFiles((prevFiles) => [...prevFiles, ...newFiles]);
+    console.log({ selectedFiles });
+  };
```
Comment (Chrissydrx) [inline src/pages/EmployeeCreation/components/UploadBox.jsx]: This seems like you can upload multiple files at once, but you can only chose one file in the picker

File: src/pages/EmployeeCreation/components/UploadBox.jsx [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  const handleFileDelete = (fileNameToDelete) => {
+    setSelectedFiles((prevFiles) =>
+      prevFiles.filter((file) => file.name !== fileNameToDelete)
+    );
+  };
+
+  return (
+    <Box>
+      {selectedFiles.map((file) => (
+        <Box key={file.name} display="flex" alignItems="center" mb={1}>
```
Comment (Chrissydrx) [inline src/pages/EmployeeCreation/components/UploadBox.jsx]: Use as key an uuid

File: src/pages/EmployeeCreation/components/UploadedFile.jsx [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+*/
+function UploadedFile({ fileName, onDelete }) {
+  const deleteFile = () => {
+    onDelete(fileName);
+  };
+
+  return (
+    <Box display="flex" alignItems="center" flexDirection="row" mb={1}>
+      <InsertDriveFile fontSize="small" style={{ marginRight: 8 }} />
+      <Typography variant="subtitle1">Selected File: {fileName}</Typography>
```
Comment (Chrissydrx) [inline src/pages/EmployeeCreation/components/UploadedFile.jsx]: Set default for filename

File: src/pages/EmployeeCreation/components/UploadedFile.jsx [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+import DeleteIcon from '@mui/icons-material/Delete';
+import PropTypes from 'prop-types'
+
+
+/**
+@returns {JSX.Element} The File Component for Employee creation page to display uploaded Files.
+*/
+function UploadedFile({ fileName, onDelete }) {
+  const deleteFile = () => {
+    onDelete(fileName);
```
Comment (Chrissydrx) [inline src/pages/EmployeeCreation/components/UploadedFile.jsx:12]: Set default

File: src/pages/EmployeeCreation/components/UploadBox.jsx [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+    const files = event.target.files;
+    const newFiles = Array.from(files);
+
+    setSelectedFiles((prevFiles) => [...prevFiles, ...newFiles]);
+    console.log({ selectedFiles });
+  };
+
+  const handleFileUpload = () => {
+    document.getElementById("upload-file").click();
+  };
```
Comment (Chrissydrx) [inline src/pages/EmployeeCreation/components/UploadBox.jsx]: Does this click the upload-file input field?

**Review by Chrissydrx — CHANGES_REQUESTED** [review]

File: src/pages/EmployeeCreation/EmployeeCreationPage.jsx
Comment (Arkcorner) [reply src/pages/EmployeeCreation/EmployeeCreationPage.jsx:1]: The Layout for small devices has been fixed

Comment (Arkcorner) [reply src/pages/EmployeeCreation/EmployeeCreationPage.jsx:1]: I now use the Prettier settings

File: src/pages/EmployeeCreation/EmployeeCreationPage.jsx (hunk 1 above)
Comment (Arkcorner) [reply src/pages/EmployeeCreation/EmployeeCreationPage.jsx]: theres now spaces between the functions

File: src/pages/EmployeeCreation/EmployeeCreationPage.jsx (hunk 2 above)
Comment (Arkcorner) [reply src/pages/EmployeeCreation/EmployeeCreationPage.jsx]: the sizes are now both dependent on the width of the screen and i didnt really see a convention from us for this so i tried to make it just look good

File: src/pages/EmployeeCreation/components/UploadBox.jsx (hunk 5 above)
Comment (Arkcorner) [reply src/pages/EmployeeCreation/components/UploadBox.jsx]: did

File: src/pages/EmployeeCreation/EmployeeCreationPage.jsx
Comment (Chrissydrx) [reply src/pages/EmployeeCreation/EmployeeCreationPage.jsx:1]: image
Still not completely fixed for width 900px

File: src/pages/EmployeeCreation/components/UploadBox.jsx (hunk 3 above)
Comment (Chrissydrx) [reply src/pages/EmployeeCreation/components/UploadBox.jsx]: Still not fixed

File: src/pages/EmployeeCreation/components/UploadedFile.jsx (hunk 7 above)
Comment (Chrissydrx) [reply src/pages/EmployeeCreation/components/UploadedFile.jsx:12]: or at least a fallback

**Review by Chrissydrx — CHANGES_REQUESTED** [review]

File: src/pages/EmployeeCreation/EmployeeCreationPage.jsx
Comment (Arkcorner) [reply src/pages/EmployeeCreation/EmployeeCreationPage.jsx:1]: fixed

File: src/pages/EmployeeCreation/components/UploadBox.jsx (hunk 4 above)
Comment (Arkcorner) [reply src/pages/EmployeeCreation/components/UploadBox.jsx]: Yes files need to be uploaded one at a time array is just to allow multiple uploads one after the other

File: src/pages/EmployeeCreation/components/UploadedFile.jsx (hunk 7 above)
Comment (Arkcorner) [reply src/pages/EmployeeCreation/components/UploadedFile.jsx:12]: default should be set

File: src/pages/EmployeeCreation/components/UploadBox.jsx (hunk 8 above)
Comment (Arkcorner) [reply src/pages/EmployeeCreation/components/UploadBox.jsx]: yes a straight up on click doesnt seem to work and i got this from coPilot

File: src/pages/EmployeeCreation/components/UploadedFile.jsx (hunk 6 above)
Comment (Arkcorner) [reply src/pages/EmployeeCreation/components/UploadedFile.jsx]: did

File: src/pages/EmployeeCreation/EmployeeCreationPage.jsx
Comment (Arkcorner) [reply src/pages/EmployeeCreation/EmployeeCreationPage.jsx:1]: Screenshot 2023-12-13 at 17 24 38

File: src/pages/EmployeeCreation/components/UploadBox.jsx (hunk 3 above)
Comment (Arkcorner) [reply src/pages/EmployeeCreation/components/UploadBox.jsx]: fixed

**Comment by j0world** [comment]
If you open Employee overview on a smaller screen you can see that it doesn't open at the top. Others don't do that either but if you want to fix it you can ask Uzoma on how to do that. 
Only being able to upload a single file at a time is already mentioned by Christian.
I can create a new employee without any information which isn't as bad because there's no logic behind the handleSubmit() anyway but if you're ambitious, that shouldn't be too difficult to change.

File: src/pages/EmployeeCreation/EmployeeCreationPage.jsx [hunk 9]
```diff
@@ -0,0 +1,179 @@
...
+            flexDirection: 'column',
+          }}
+        >
+          <Box maxWidth='50%'>
+            <Box>
+              <Avatar
+                sx={{ width: '10vw', height: '10vw', variant: 'circular' }}
+              />
+              <Fab
+                position='absolute'
```
Comment (j0world) [inline src/pages/EmployeeCreation/EmployeeCreationPage.jsx:90]: The profile picture scales but the plus button doesn't.

**Review by j0world — CHANGES_REQUESTED** [review]
If I add too much stuff, either in Files or Extra info, the Create Employee button lingers at the bottom and can't be moved away.
The same for the profile picture but with the top.

File: src/pages/EmployeeCreation/EmployeeCreationPage.jsx [hunk 10]
```diff
@@ -0,0 +1,179 @@
...
+                onChange={handleEmployeeEmailChange}
+                error={employeeEmailError}
+                helperText={
+                  employeeEmailError ? 'Please enter a valid email address' : ''
+                }
+              />
+            </Box>
+            <Box>
+              <TextField
+                label='Phone number'
```
Comment (j0world) [inline src/pages/EmployeeCreation/EmployeeCreationPage.jsx]: Please change to "Phone Number" according to title case

File: src/pages/EmployeeCreation/EmployeeCreationPage.jsx [hunk 11]
```diff
@@ -0,0 +1,179 @@
...
+                sx={{ minWidth: '20vw' }}
+                value={employeeAddress}
+                onChange={handleEmployeeAddressChange}
+              />
+            </Box>
+            <Box sx={{ flexDirection: 'column' }}>
+              <TextField
+                margin='normal'
+                type='submit'
+                label='Extra info'
```
Comment (j0world) [inline src/pages/EmployeeCreation/EmployeeCreationPage.jsx]: Please change to "Extra Info" according title case

File: src/pages/EmployeeCreation/EmployeeCreationPage.jsx (hunk 9 above)
Comment (j0world) [reply src/pages/EmployeeCreation/EmployeeCreationPage.jsx:90]: width changes now, height doesn't seem to scale correctly

Comment (Arkcorner) [reply src/pages/EmployeeCreation/EmployeeCreationPage.jsx:90]: the + in the fab hinders it from scaling correctly for all sizes but in my testing it was ok for the 900 by 900 px and full screen size I'm not sure how to make it completely dynamic

**Review by j0world — APPROVED** [review]

File: src/pages/EmployeeCreation/components/UploadBox.jsx
Comment (j0world) [inline src/pages/EmployeeCreation/components/UploadBox.jsx:1]: I clicked random things out of curiosity, please make it possible to upload multiple files and I shall also willingly approve

**Review by j0world — CHANGES_REQUESTED** [review]

File: src/pages/EmployeeCreation/components/UploadBox.jsx (hunk 4 above)
Comment (Arkcorner) [reply src/pages/EmployeeCreation/components/UploadBox.jsx]: Fixed the Input component of material UI mad this way harder as it didn´t have the multiple property defined so i switched to a normal input field to more suit the needs of our website as its hidden anyway

File: src/pages/EmployeeCreation/components/UploadBox.jsx
Comment (Arkcorner) [reply src/pages/EmployeeCreation/components/UploadBox.jsx:1]: did

Comment (j0world) [reply src/pages/EmployeeCreation/components/UploadBox.jsx:1]: indeed works now. 
If I upload a document with too long of a title however, it messes with your padding
Screenshot 2024-01-23 at 18 47 44

**Review by j0world — APPROVED** [review]

**Review by Chrissydrx — APPROVED** [review]
I trust that you guys did a good job. I didn't check the PR, but because I don't have time in the next days, I will approve this, so that I don't block the team

**Pull request merged**